### PR TITLE
Fixed linker error in Visual Studio

### LINKED ISSUE #21161: Problem compiling with Visual Studio
Using Visual Studio 2015 on Windows 7 can't compile latest SDL tiles version.

Here is the log with /verbose (today I learn what that is)
[Cataclysm.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/1043248/Cataclysm.txt)

After spending all this day looking what the linker error were, and after manually checking commits(I discovered later that I could have used git bisect) I found out that this commit  https://github.com/CleverRaven/Cataclysm-DDA/commit/dbf94ea32432320fc874237d93965c069fb674f3 is the one causing the problem.

I would have spotted this problem earlier if I were more used to sync my local repo, sorry about that.

### PR DESCRIPTION
**Pull request by AlecWhite**
Fixes #21161 allowing Visual Studio to compile the latest versions.

Thanks to @ufft47 for the original fix and @Graywolfe813 for pointing out why the fix didn't work.

### CODE REVIEW

File: src/string_id_null_ids.cpp [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
-// These are centralized in this file because they must appear in one and only one compilation unit,
-// and it's problematic to define them in the respective cpp files for each class.
-// Very repitious, so define them with a macro.
-#define MAKE_NULL_ID( type, ... ) \
-class type; \
-template<> const string_id<type> &string_id<type>::NULL_ID() { \
-    static string_id<type> id = string_id<type>( __VA_ARGS__ ); \
-    return id; \
-}
+#define MAKE_NULL_ID( type, ... )  class type;  template<> const string_id<type> &string_id<type>::NULL_ID() {  static string_id<type> id = string_id<type>( __VA_ARGS__ );  return id; }
```
Comment (kevingranade) [inline src/string_id_null_ids.cpp]: commented and readable -> uncommented and unreadable
What makes you think this is an ok change? :/

Comment (AlecWhite) [inline src/string_id_null_ids.cpp]: I don't have the slighliest idea of how anything in this .cpp works. I barely know enough about preprocessors to make guards and include whatever header I need.
But after a week of not being able to compile, I just tried by whatever means to make it work, hence copypasting things I don't understand and asking in IRC for help. 
And to remark by total lack of knowledge and experience I just applied another fix by @Graywolfe813.

**Comment by AlecWhite** [comment]
And because it never rains but it pours. @Qrox pointed out that the array of size 0 error that stopped VS to from compiling was introduced by https://github.com/CleverRaven/Cataclysm-DDA/pull/21173.

I feel like I'm complaining a lot and opening a lot of issues(switch to SDL2 2.0.5, this null string error, and now initialized array of size 0).
Should I open a new issue regarding that?

**Comment by kevingranade** [comment]
Yes please open a new issue for the array thing.
These reports are valuable, no need to feel bad about opening them.

**Pull request merged**